Source/WebCore/platform/mac/ScrollbarThemeMac.mm

```diff
@@ -557,9 +557,21 @@ static void scrollerImpPaint(NSScrollerImp *scrollerImp, bool enabled)
 
     GraphicsContextStateSaver stateSaver(context);
     context.clip(damageRect);
-    context.translate(scrollbar.frameRect().location());
-    LocalCurrentGraphicsContext localContext(context);
-    scrollerImpPaint(scrollbarMap().get(&scrollbar).get(), scrollbar.enabled());
+
+    auto scrollbarRect = scrollbar.frameRect();
+    if (context.platformContext()) {
+        context.translate(scrollbarRect.location());
+        LocalCurrentGraphicsContext localContext { context };
+        scrollerImpPaint(scrollbarMap().get(&scrollbar).get(), scrollbar.enabled());
+    } else {
+        auto imageBuffer = [&] {
+            auto buffer = ImageBuffer::create(scrollbarRect.size(), RenderingPurpose::Unspecified, scrollbar.deviceScaleFactor(), DestinationColorSpace::SRGB(), PixelFormat::BGRA8);
```

Comment:
Should we use `context.createImageBuffer()` to abstract away some of the arguments?